packages/js/src/ui/context/LocalizationContext.tsx

```diff
@@ -1,17 +1,47 @@
 import { Accessor, createContext, createMemo, ParentProps, useContext } from 'solid-js';
-import { defaultLocalization } from '../config/defaultLocalization';
-import type { Localization, LocalizationKey } from '../types';
+import { defaultLocalization, dynamicLocalization } from '../config/defaultLocalization';
 
-type LocalizationValue<K extends LocalizationKey> = (typeof defaultLocalization)[K];
-type LocalizationParams<K extends LocalizationKey> = LocalizationValue<K> extends (args: infer P) => any
-  ? P
+type DefaultLocalizationKey = keyof typeof defaultLocalization;
+// eslint-disable-next-line @typescript-eslint/ban-types
+export type LocalizationKey = DefaultLocalizationKey;
+
+export type FunctionLocalizationKey = {
+  [K in DefaultLocalizationKey]: (typeof defaultLocalization)[K] extends (...args: any[]) => any ? K : never;
+}[DefaultLocalizationKey];
+
+export type StringLocalizationKey = {
+  [K in DefaultLocalizationKey]: (typeof defaultLocalization)[K] extends string ? K : never;
+}[DefaultLocalizationKey];
+
+export type LocalizationValueOverride<K extends LocalizationKey> = K extends FunctionLocalizationKey
+  ? (...args: Parameters<(typeof defaultLocalization)[K]>) => ReturnType<(typeof defaultLocalization)[K]>
+  : K extends StringLocalizationKey
+  ? string
+  : string; // Allow any string value for arbitrary keys
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** I don't see this type is used anywhere, remove? And I feel like it's not correct because K would be an object type and you meant to get the key of it. Don't forget to remove the above two types too: `FunctionLocalizationKey, StringLocalizationKey`

Reply:
Same as above. Leftover from the other attempt. Thanks!